nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -652,7 +652,11 @@ def forward(
 
                     # speech_logits_list will be used in loss calculation (parallel output)
                     speech_logits_list = []
-                    if self.seq_pattern in ["parallel", "delay_parallel"]:
+                    speech_logits = torch.zeros(
+                        [*token_logits.shape[:-1], self.speech_codebook_size, speech_layers],
+                        device=token_logits.device,
+                    )
+                    if self.seq_pattern in ["parallel", "delay_parallel"] and torch.count_nonzero(speech_mask) > 0:
```

Comment:
same issue here, if we train jointly on ASR and TTS task.

Reply:
same.